### PR TITLE
Python3 and 2 compatibility + latest dev

### PR DESCRIPTION
**Pull request by pzhokhov**
1. Enable tests with python 3
2. Fix python 2/3 compatibility issues
3. Latest development branch 
(it has completion service for which I still want docs from you, @jasonzliang :), but also a bunch of other handy features to help with python2/3 compatibility, hence I decided to merge it in as well)

### CODE REVIEW

File: studio/app.yaml [hunk 1]
```diff
@@ -1,7 +1,11 @@
-runtime: python27
```
Comment (nieoh) [inline studio/app.yaml]: This probably needs to be revisited to reflect the policy on which python to use in gcloud.

File: studio/completion_service/completion_service.py [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+import uuid
+import logging
+import time
+import pickle
+import tempfile
+import signal
+import re
+import six
+
+from studio import runner, model, fs_tracker
```
Comment (nieoh) [inline studio/completion_service/completion_service.py]: I feel like we should be consistent with our relative imports (`from studio import runner` vs. `from . import runner`)

Comment (pzhokhov) [reply studio/completion_service/completion_service.py]: but that's in a different folder. We could use `from .. import runner`, but we are already doing 
`from studio import something` in the tests

File: studio/app.yaml (hunk 1 above)
Comment (pzhokhov) [reply studio/app.yaml]: yeah, to be frank, I have not re-deployed the gae app; so I don't know if that part will work. But we don't have such policy just yet, if I understand correctly :) Using python3 may be helpful if we try to integrate the hosted UI with python notebooks in some way; but for now I am quite ambivalent on that.

**Pull request merged**